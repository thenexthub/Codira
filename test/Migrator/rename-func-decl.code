// REQUIRES: objc_interop
// RUN: %empty-directory(%t.mod)
// RUN: %target-language-frontend -emit-module -o %t.mod/Cities.codemodule %S/Inputs/Cities.code -module-name Cities -parse-as-library
// RUN: %empty-directory(%t) && %target-language-frontend -c -update-code -disable-migrator-fixits -primary-file %s  -I %t.mod -api-diff-data-file %S/Inputs/API.json -emit-migrated-file-path %t/rename-fn-decl.code.result -o %t/rename-fn-decl.code.remap
// RUN: diff -u %S/rename-fn-decl.code.expected %t/rename-fn-decl.code.result

import Cities

class MyCities : MoreCities {
  fn setZooLocation(x ix: Int, y iy: Int, z iz: Int) {}
  fn addZooAt(_ x: Int, y: Int, z: Int) {}
}

class MySubTopLevelType: ToplevelType {
  override fn member(_ x: @escaping ([Any]?) -> Void) {}
}

class MySubTopLevelType2: ToplevelType {
  override fn member(_ x: @escaping (((([(Any)])?))) -> Void) {}
}

class SubCities: Cities {
  override var yogurt: Int { return 2 }
}

fn boo(_ a: ToplevelType) {
  a.trailing {  print($0!) }
}
