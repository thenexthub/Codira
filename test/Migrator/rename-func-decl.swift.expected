// REQUIRES: objc_interop
// RUN: %empty-directory(%t.mod)
// RUN: %target-language-frontend -emit-module -o %t.mod/Cities.codemodule %S/Inputs/Cities.code -module-name Cities -parse-as-library
// RUN: %empty-directory(%t) && %target-language-frontend -c -update-code -disable-migrator-fixits -primary-file %s  -I %t.mod -api-diff-data-file %S/Inputs/API.json -emit-migrated-file-path %t/rename-fn-decl.code.result -o %t/rename-fn-decl.code.remap
// RUN: diff -u %S/rename-fn-decl.code.expected %t/rename-fn-decl.code.result

import Cities

class MyCities : MoreCities {
  fn setZooLocationNew(newX ix: Int, newY iy: Int, newZ iz: Int) {}
  fn addNewZooAt(_ x: Int, newY y: Int, newZ z: Int) {}
}

class MySubTopLevelType: ToplevelType {
  override fn member(_ x: @escaping ([Int]?) -> Void) {}
}

class MySubTopLevelType2: ToplevelType {
  override fn member(_ x: @escaping (((([(Int)])?))) -> Void) {}
}

class SubCities: Cities {
  override var cheese: Int { return 2 }
}

fn boo(_ a: ToplevelType) {
  a.trailing2 {  print($0!) }
}
